packages/keyring-controller/src/KeyringController.ts

```diff
@@ -523,16 +531,24 @@ function isSerializedKeyringsArray(
  * @param keyring - The keyring to display.
  * @returns A keyring display object, with type and accounts properties.
  */
-async function displayForKeyring(
-  keyring: EthKeyring<Json>,
-): Promise<{ type: string; accounts: string[] }> {
+async function displayForKeyring(keyring: EthKeyring<Json>): Promise<{
+  type: string;
+  accounts: string[];
+  typeIndex: number;
+  id: string;
+}> {
   const accounts = await keyring.getAccounts();
+  const { opts } = keyring as EthKeyring<Json> & {
+    opts: { typeIndex: number; id: string };
+  };
 
   return {
     type: keyring.type,
     // Cast to `string[]` here is safe here because `accounts` has no nullish
     // values, and `normalize` returns `string` unless given a nullish value
     accounts: accounts.map(normalize) as string[],
+    typeIndex: opts?.typeIndex,
+    id: opts?.id,
```

Comment:
This may be undefined but the types changed in the rest of the controller suggest that these are always defined. This could create unexpected scenarios